src/app.rs

```diff
@@ -524,7 +523,7 @@ impl cosmic::Application for CosmicLauncher {
             DbusActivationDetails::Activate => {
                 if self.active_surface || self.wait_for_result {
                     return self.hide();
-                } else if self.last_hide.elapsed().as_millis() > 100 {
+                } else {
```

Comment:
Why remove this? It is a hack, but it prevents the launcher from reappearing after closing it by pressing the launcher button on the panel?

Reply:
Makes sense. I thought it was related to service somehow, like don't query if it was quickly reopened. I will readd it